### ISSUE TITLE
What about a tool for participants without privilege to signal their incomprehension?

### ISSUE THREAD

**Issue by philardine** [tableaunoir/tableaunoir#118]
Dear François,
As I was writing my last comment for issue #117, I realized something unrelated to the topic but nevertheless interesting.

From my point of view, when students have no privileges their mouse pointers are only relevant on the following points:
1. I can verify that I am not preaching in the desert >:-(
2. they can alert me on some problem in my presentation by vibrating around some region with their mouse

Well I can negotiate with my ego for the first point (especially since I can read the number of participants).

But I believe that the second point could be rather useful... Perhaps it would be nice if participants with no privilege had a unique drawing tool to highlight difficult points on the board, thus manifesting their incomprehension? Kind of rendering in a flashy way some portion of the board...
What do you think?

**Comment by francoisschwarzentruber** [comment]
Dear philardine,
thank you for your proposal. Very interesting.

A) Users can already ask questions in the tool (in the menu shared, a bit hidden so that they do not flood a course) But I guess that is a bit useless since we may use another tool for chatting, discussing.

B) We could decide not to implement that feature, by using that users that move have questions? I like this because it costs nothing :).

C) We could implement that "by default". If you cannot write on the board, the effect of drawing is automatically a laser pointer. The issue is that laser pointer may be useful also for people that can write. I like this option because the interface keeps simple (no extra button).

D) We could implement a new tool "laser pointer" with a new button. I tend not to like this option because maybe the interface of tableaunoir should remain childish/simple/stupid, but maybe I am wrong.

And you? What is your preferred option? Do you have other ideas in mind?

**Comment by philardine** [comment]
Dear François,
Thank you for showing interest in my suggestion.

I like your point C, I suggest that 
1. mouse moves of unprivileged participants are not sent to the server until they happen to be clics,
2. drag gestures activate the default laser tool.

I guess point 1. would further improve your « naïve improvement », wouldn't it?

IMHO:

A) this does not seem to me to be the right way to manage the problem: we are actually mixing two different tools
B) this solution could unnecessarily consume time on the server side (even with your last naïve improvement)
D) I agree with your argument about the simple interface... but read further ;-)

But this laser tool may also be useful for teachers!
Wouldn't it be interesting to have a button to switch between pencil and laser pointer (as the eraser is) when you are able to draw on the board?
So the laser tool would be the default writing for unprivileged users, and could be switched for privileged ones.

**Comment by francoisschwarzentruber** [comment]
Dear Philardine,

thank you for your interesting reply. I have few comments.

Concerning A), do you think the functionality of sending messages/questions should be abandoned/removed? (I remember that other persons told me that it is useless) I think that it could be removed and that it was a bad idea to propose the option to send questions in tableaunoir.

Concerning the mouse moves, I think I feel reassured when I see them: I can see that the connection is working, that others are doing something, I "feel" the presence of other users. If they can write, I will see in advance where they are going to write, etc.

I will think about it.

**Comment by philardine** [comment]
Well, I have to admit that I never use the question tool, nor my students. This is undoubtedly related to the fact that we never use tableaunoir as a standalone tool, but in combination with some video conferencing tool.

Still, I don't vote for an amputation: as it is today, I understand this tool more as a request for attention than a real chat or tool for asking questions... maybe it could be useful one day?

**Comment by francoisschwarzentruber** [comment]
Maybe... but I have some doubts... Initially, the idea was that students ask questions and that it is "questions on the bottom of the screen" like on in some TV shows. But finally, I think that is a bad idea. :)

**Comment by philardine** [comment]
For sure, implementation of a laser tool would drastically decrease the interest of this « questions line »...

**Comment by francoisschwarzentruber** [comment]
I agree... :) And given the interest was already very low, it will be infinitesimal!

**Comment by Naereen** [comment]
My opinions are twofold:
- this LASER pointer is maybe useful but not required ;
- the "tiny chat" for asking question is awesome, and I love it, even if didn't see any real use so far.

**Comment by francoisschwarzentruber** [comment]
I am sorry that the laser pointer was never implemented. It was by lack of time and also because I never needed that functionality. My students just ask questions vocally, and just say "on the top right corner of the board, what is x?". Or I saw the pointers of others, and it was sufficient.
